docs/preflight.md

```diff
@@ -0,0 +1,7 @@
+## Preflight Checklist
+
+This checklist is to be ran prior to kicking off the manual test plan.
```

Comment:
nit: `prior to cutting the release branch`
otherwise we have to backport all these changes from master.